### PR TITLE
Chemical specific heats!

### PR DESCRIPTION
**Pull request by silicons**
:cl:
experimental: Chemistry now has something called specific heat. For the uninitiated, this is the thermal capacity of the reagent, much like how some atmospherics gases have different thermal capacities, making them able to absorb/release more energy per degree of temperature change!
rscadd: Plasma specific heat is 500J/K*unit, everything else is 200
/:cl:
# Currently default is 200 and plasma has 500. Thoughts on other chemicals?

image
^ "for dummies" version

### CODE REVIEW

**Comment by kevinz000** [comment]
@Iamgoofball i'll replace a few thousand ids with long typepaths when i have a few hours to kill and this isn't that much laggier tbh

**Comment by Ordonis** [comment]
Yes, this will allow my physics autism to run wild by attaching values to each and every reagent

**Comment by Tlaltecuhtli** [comment]
does it mean ice melts

**Comment by kevinz000** [comment]
image
think i fucked something up

**Comment by kevinz000** [comment]
oh i see i forgot new_total in the equation for new temp

**Comment by ExcessiveUseOfCobblestone** [comment]
what would be the practical effects of this besides being able to make meth explosion inside someone?

**Comment by arsserpentarium** [comment]
Fabolous.

**Comment by Cyberboss** [comment]
>tfw you fully agree with a goof comment

**Comment by kevinz000** [comment]
opkay tested and working

**Comment by zamolxius** [comment]
for what purpose

**Comment by kevinz000** [comment]
eh i'm hoping we can have some unique chemicals with lower/hgiher specific heat than usual but if we feel like it's unnecessary just close it i'm mostly doing it because there isn't a lot of performance hit and it'd allow for fun implementations down the line.

**Comment by davethwave** [comment]
It probably won't be too hard to make the specific heat realistic

**Comment by ExcessiveUseOfCobblestone** [comment]
sir if you aren't adding implementation this is just creating things for the luls

**Comment by davethwave** [comment]
Could be useful for heating/cooling people down using chems with high specific heat I guess. If this does get merged I guess I could make a chem with large specific heat to see if people find a use for it.

**Comment by kevinz000** [comment]
^ it's for things like that.

**Comment by optimumtact** [comment]
I don't know if this makes sense outside of an actual chemistry overhaul

**Review by SpaceManiac — APPROVED** [review]
Not 100% sure this is necessary as a feature or worth the >kevinzcode maintenance burden but the code looks good and works in testing

**Comment by Iamgoofball** [comment]
>merged 19 hours ago
>cant autorevert

can someone please fire kevin for doing stupid crap like this

**Comment by optimumtact** [comment]
\>not willing to put any effort into reverting

good riddance lad

**Comment by Winterous** [comment]
THIS.
THIS IS THE CAUSE OF THIS ISSUE! https://github.com/tgstation/tgstation/issues/38216
HE EVEN PREDICTED IT IN A CODE COMMENT!

File: code/modules/reagents/chemistry/holder.dm [hunk 1]
```diff
@@ -526,19 +526,54 @@
...
 
+	var/datum/reagent/D = GLOB.chemical_reagents_list[reagent]
+	if(!D)
+		WARNING("[my_atom] attempted to add a reagent called '[reagent]' which doesn't exist. ([usr])")
+		return FALSE
+
+ 	update_total()
+	var/cached_total = total_volume
+	if(cached_total + amount > maximum_volume)
+		amount = (maximum_volume - cached_total) //Doesnt fit in. Make it disappear. Shouldnt happen. Will happen.
```
Comment (Winterous) [inline code/modules/reagents/chemistry/holder.dm]: THIS BIT RIGHT HERE!

Comment (SpaceManiac) [reply code/modules/reagents/chemistry/holder.dm]: This behavior isn't new in this PR

Comment (Winterous) [reply code/modules/reagents/chemistry/holder.dm]: Then why is it happening? q-q

**Pull request merged**